geo/src/main/java/org/odk/collect/geo/selection/SelectionMapFragment.kt

```diff
@@ -209,14 +205,37 @@ class SelectionMapFragment(
         )
     }
 
-    private fun restoreZoomFromPreviousState(state: Bundle) {
+    private fun initializeViewport() {
+        previousState?.let {
+            if (restoreZoomFromPreviousState(it)) {
+                viewportInitialized = true
+            }
+        }
+
+        if (!viewportInitialized) {
+            itemsByFeatureId.filter { it.value.selected }.map { it.key }.firstOrNull()?.let {
+                onFeatureClicked(it)
```

Comment:
I have some trouble understanding how the selected item is kept track of. It looks like if there's an item selected in the form that comes in as one of the items in the list having a selected field set. Then this initial call on `onFeatureClicked` saves the id of that item in the viewmodel. And then the `selected` field on `MappableSelectItem` objects is entirely ignored and stale. This makes me uneasy. Can the viewmodel not be created with the selectedid and the `selected` property be removed on `MappableSelectItem`?